codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -140,6 +143,100 @@ public void onPriorityTreeParentChanging(Http2Stream stream, Http2Stream newPare
                 }
             }
         });
+
+        if (listener == null) {
```

Comment:
maybe instead of doing this, we just internally set a `noop` listener so we don't have to check for `listener == null` everywhere? I don't think the performance difference is worth the extra complexity.